src/p2p/peer.rs

```diff
@@ -99,38 +93,20 @@ impl Peer {
             NetMessage::MempoolMessage(mempool_msg) => {
                 debug!("Received new mempool net message {:?}", mempool_msg);
                 self.bus
-                    .sender::<NetInput<MempoolNetMessage>>()
+                    .sender::<MempoolNetMessage>()
                     .await
-                    .send(NetInput::new(mempool_msg.clone()))
+                    .send(mempool_msg.clone())
```

Comment:
Clone not needed I think
```suggestion
                    .send(mempool_msg)
```